locations/spiders/ikks.py

```diff
@@ -0,0 +1,45 @@
+import json
+import re
+
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS_FR, OpeningHours
+
+
+class IkksSpider(scrapy.Spider):
+    name = "ikks"
+    item_attributes = {"brand": "IKKS", "brand_wikidata": "Q3146711"}
+    allowed_domains = ["stores.ikks.com"]
+    start_urls = [
+        "https://stores.ikks.com/mobify/proxy/ocapi/s/IKKS_COM/dw/shop/v21_3/stores?latitude=24.3356791&longitude=23.769953599999997&max_distance=20000&delivery_method=delivery&client_id=1c53da3a-3640-4a31-adad-0f43be6c0904&start=0&count=200"
+    ]
+
+    def parse(self, response):
+        if not response.json().get("count"):
+            return
+        for data in response.json().get("data"):
+            if "IKKS" in data.get("c_commercialSign"):
+                item = DictParser.parse(data)
+                oh = OpeningHours()
+                for day in json.loads(data.get("c_workingSchedule")):
+                    if not day.get("amBegin"):
+                        continue
+                    if day.get("amEnd") and day.get("pmBegin"):
+                        oh.add_range(
+                            day=DAYS_FR[day.get("title")[:2]], open_time=day.get("amBegin"), close_time=day.get("amEnd")
+                        )
+                        oh.add_range(
+                            day=DAYS_FR[day.get("title")[:2]], open_time=day.get("pmBegin"), close_time=day.get("pmEnd")
+                        )
+                    else:
+                        oh.add_range(
+                            day=DAYS_FR[day.get("title")[:2]], open_time=day.get("amBegin"), close_time=day.get("pmEnd")
+                        )
```

Comment:
I think these 3 conditionals are a little weird. ~"If none, return, if am and pm, add am and pm, otherwise, add am". Why not just ~"if am, add am, if pm, add pm"?

Reply:
I've not tested, but something like:

```suggestion
                    if day.get("amBegin") and day.get("amEnd"):
                        oh.add_range(
                            day=DAYS_FR[day.get("title")[:2]], open_time=day.get("amBegin"), close_time=day.get("amEnd")
                        )
                    if day.get("pmBegin") and day.get("pmEnd"):
                        oh.add_range(
                            day=DAYS_FR[day.get("title")[:2]], open_time=day.get("pmBegin"), close_time=day.get("pmEnd")
                        )
```

Reply:
Ex:
```
mar-ven : 09:30 - 12:30 / 15:00 - 19:00
```

Reply:
Ah, I hadn't predicted Ex3. Looks like your code will miss Ex2 though?

Reply:
yes !